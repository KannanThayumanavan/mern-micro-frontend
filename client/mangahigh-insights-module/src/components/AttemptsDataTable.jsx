import React from 'mangahigh-libs/react';
import Table from 'mangahigh-components/Table';
import PropTypes from 'prop-types';

const populateRowsToDisplay = (attempts) => attempts.map((attempt, index) => [
	(index + 1).toString(),
	new Date(attempt.date_and_time).toUTCString(),
	attempt.score.toString(),
	attempt.total_questions.toString(),
]);

const AttemptsDataTable = ({ attempts }) => {	
	const tableData = {
	  header: ['#', 'Date and time of play', 'Score', 'Total Questions'],
	  rows: [...populateRowsToDisplay(attempts)],
	};
	return (
		<div className="row">
			<div className="col-md-12">
				<h4 className="text-left">Historical Results <small>(latest on top)</small></h4>
				<Table tableData={tableData} />
			</div>
		</div>
	);
}

AttemptsDataTable.propTypes = {
	attempts: PropTypes.arrayOf(PropTypes.shape({
		date_and_time: PropTypes.string,
		score: PropTypes.number,
		total_questions: PropTypes.number,
	})).isRequired
};

export default AttemptsDataTable;