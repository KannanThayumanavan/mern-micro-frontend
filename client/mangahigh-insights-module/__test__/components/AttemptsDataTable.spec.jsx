import React from 'react';
import { shallow, mount } from 'enzyme';
import AttemptsDataTable from '../../src/components/AttemptsDataTable';

jest.mock('mangahigh-libs/react', () => jest.requireActual('react'),
	{ virtual: true }
);

jest.mock('mangahigh-components/Table', () => () => 'Table'			
, { virtual: true });

describe('Test AttemptsDataTable Component', () => {
	const attempts = [
		{
			"_id":"6161b01c74f86b0917b418c8",
			"date_and_time":"2021-10-09T15:07:07.763Z",
			"score":3,
			"total_questions":3
		},
		{
			"_id":"6161b01c74f86b0917b418b6",
			"date_and_time":"2021-10-09T15:37:07.763Z",
			"score":1,
			"total_questions":3
		}
	];
	it('should render correctly with required props', () => {
		const renderedModule = shallow(<AttemptsDataTable attempts={attempts} />);
    expect(renderedModule).toMatchSnapshot();
	});
});