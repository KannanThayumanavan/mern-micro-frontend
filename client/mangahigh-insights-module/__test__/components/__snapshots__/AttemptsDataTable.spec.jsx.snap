// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AttemptsDataTable Component should render correctly with required props 1`] = `
<div
  className="row"
>
  <div
    className="col-md-12"
  >
    <h4
      className="text-left"
    >
      Historical Results 
      <small>
        (latest on top)
      </small>
    </h4>
    <Component
      tableData={
        Object {
          "header": Array [
            "#",
            "Date and time of play",
            "Score",
            "Total Questions",
          ],
          "rows": Array [
            Array [
              "1",
              "Sat, 09 Oct 2021 15:07:07 GMT",
              "3",
              "3",
            ],
            Array [
              "2",
              "Sat, 09 Oct 2021 15:37:07 GMT",
              "1",
              "3",
            ],
          ],
        }
      }
    />
  </div>
</div>
`;
